### PR TITLE
Initial support for documentation versions

### PR DESCRIPTION
**Pull request by scsides**
Add ability to get to multiple versions of the ISIS documentation

## Description
Multiple versions of ISIS documentation can now live side-by-side. New versions will be placed in a dir named from the ISIS version.

## Motivation and Context

There have been many requests for old version of the docs.

## How Has This Been Tested?

This has been locally deployed and all of the pages hand tested

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I have read and agree to abide by the [Code of Conduct](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/Code-Of-Conduct.md)
- [X] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [X] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [ ] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [X] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/cmake/BuildDocs.cmake [hunk 1]
```diff
@@ -66,22 +66,30 @@ endfunction(copy_app_docs_info)
...
+  copy_folder(${docBuildFolder}/w3c    ${docInstallFolder}/${docVersion})
+
   # These folders are populated inside "build_documents_folder"
 
-  # Create index.html file
-  execute_process(COMMAND ${XALAN} ${XALAN_VALIDATE_OPTION} ${XALAN_PARAM_OPTION} menuPath \"\" ${XALAN_OUTFILE_OPTION} ${docInstallFolder}/index.html ${XALAN_INFILE_OPTION} ${docBuildFolder}/build/homepage.xml ${XALAN_XSL_OPTION} ${docBuildFolder}/build/main.xsl)
+  # Create the main index.html file aka the home page
+  execute_process(COMMAND ${XALAN} ${XALAN_VALIDATE_OPTION} ${XALAN_PARAM_OPTION} menuPath \"${docVersion}/\" ${XALAN_OUTFILE_OPTION} ${docInstallFolder}/index.html ${XALAN_INFILE_OPTION} ${docBuildFolder}/build/homepage.xml ${XALAN_XSL_OPTION} ${docBuildFolder}/build/main.xsl)
+  # Create a second main page inside the version numbered area for when the page above gets overwritten with a new version 
+  execute_process(COMMAND ${XALAN} ${XALAN_VALIDATE_OPTION} ${XALAN_PARAM_OPTION} menuPath \"\" ${XALAN_OUTFILE_OPTION} ${docInstallFolder}/${docVersion}/index.html ${XALAN_INFILE_OPTION} ${docBuildFolder}/build/homepage.xml ${XALAN_XSL_OPTION} ${docBuildFolder}/build/main.xsl)
```
Comment (ladoramkershner) [inline isis/cmake/BuildDocs.cmake:89]: So here, the plan is to make two identical doc builds that will get pushed up to the server. The first being the main homepage you navigate to when you go to isis.astrogeology.usgs.gov and the other being the homepage you navigate to if you click on the current version in the drop down menu?

File: isis/cmake/BuildDocs.cmake [hunk 2]
```diff
@@ -464,8 +472,10 @@ endfunction(build_object_docs)
 function(build_docs)
 
   message("Building Isis Documentation...")
+  message(" Version is: " ${VERSION})
 
   # Set up output directory and a temporary directory for building
+  set(docVersion       7.0.0)
```
Comment (ladoramkershner) [inline isis/cmake/BuildDocs.cmake]: Ignoring, but noting (for later reviews)

File: isis/cmake/BuildDocs.cmake [hunk 3]
```diff
@@ -464,8 +472,10 @@ endfunction(build_object_docs)
 function(build_docs)
 
   message("Building Isis Documentation...")
+  message(" Version is: " ${VERSION})
```
Comment (ladoramkershner) [inline isis/cmake/BuildDocs.cmake]: Where is VERSION set?

File: .gitignore [hunk 4]
```diff
@@ -51,6 +51,8 @@ install/
 
 # Unignore the documentation build
 !isis/src/docsys/build
+!isis/src/docsys/Object/build
+!isis/src/docsys/Application/build
```
Comment (ladoramkershner) [inline .gitignore:55]: Is this in preparation for splitting up the documentation builds?

Comment (scsides) [reply .gitignore:55]: No, when I updated the conf files, git warned me that these were being ignore, and they should not have been.

File: isis/cmake/BuildDocs.cmake (hunk 3 above)
Comment (scsides) [reply isis/cmake/BuildDocs.cmake]: At the top of the second from the bottom function. It's hardcoded for now, but that will change today if all goes well

File: isis/cmake/BuildDocs.cmake (hunk 1 above)
Comment (scsides) [reply isis/cmake/BuildDocs.cmake:89]: Just two identical index.html files. One in the rood of isis.astro.usgs.gov and the other in v7.0.0. This way 7.0.0 never goes away and can be fully accessed, but the root index file gets renewed every time we push a new version.  I have ideas on how we can fix this, but haven't been able to test yet.

File: isis/cmake/BuildDocs.cmake (hunk 3 above)
Comment (jessemapel) [reply isis/cmake/BuildDocs.cmake]: Use `${CMAKE_PROJECT_VERSION}` instead to get the version from ISIS.

File: isis/cmake/BuildDocs.cmake (hunk 2 above)
Comment (jessemapel) [reply isis/cmake/BuildDocs.cmake]: Same as above, just set this to `${CMAKE_PROJECT_VERSION}` and do it prior to the message above so the message can just use `${docVersion}`

**Review by jessemapel — COMMENTED** [review]
New setup seems reasonable to me. Are the changes in the doxyfiles important or just incoming changes for using a more up to date doxygen version?

**Comment by scsides** [comment]
Many new doxygen versions cause us to do a doxygen -u to get rid of all the xyz is deprecated warning. This is not a functional difference, except that now we can see real errors without wading through all junk

**Review by ladoramkershner — APPROVED** [review]

**Pull request merged**